### PR TITLE
Add initial serialization refactor through test flag

### PR DESCRIPTION
**Pull request by dgetu**
This PR includes initial work on refactoring the serialization implementation of `typespec-ts` to be a function of Modular clients and TCGC models. Incidentally, this also implements handling of recursive types and extra properties. Some tests have had their inline specs pulled into files for easier review/updating. It also introduces a dependency on `lodash`, which is already pulled in transitively through `rlc-common`. Changes should be easier to reason about when reviewing each commit individually.

Minimum requirements to expose this functionality involves populating the dynamic dispatch table in `serializeUtil` with each TCGC model type's serialization emitter callback and implementing unit tests for serialization scenarios.

### CODE REVIEW

**Review by joheredi — APPROVED** [review]
This is looking really nice. Thanks for the additional cleanup. Happy to see this also brings us a step closer to fully leveraging TCGC

File: packages/typespec-ts/test/util/emitUtil.ts [hunk 1]
```diff
@@ -440,7 +442,14 @@ export async function emitModularOperationsFromTypeSpec(
         modularCodeModel.clients[0],
         dpgContext,
         modularCodeModel,
-        false
+        false,
+        env["EXPERIMENTAL_TYPESPEC_TS_SERIALIZATION"]
```
Comment (qiaozha) [inline packages/typespec-ts/test/util/emitUtil.ts:446]: do we have to set the env EXPERIMENTAL_TYPESPEC_TS_SERIALIZATION for the serialize utils unit tests? curious if those tests are really testing something now?

File: packages/typespec-ts/src/modular/serialization/serializers.ts [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+      >
+    ) => SerializerOutput;
+  };
+  type SerializeHandlerMap = Partial<SerializeHandler<SimpleType>> &
+    SerializeHandler<
+      Exclude<(SdkType | SdkModelPropertyType)["kind"], SimpleType>
+    >;
+
+  const handlers: SerializeHandlerMap = {
+    array: placeholder,
```
Comment (qiaozha) [inline packages/typespec-ts/src/modular/serialization/serializers.ts:114]: just wonder if we could file issues or have a way to track all those places that we already know it's going to be changed? I notice you added TODO comments in other places.

File: packages/typespec-ts/test/util/emitUtil.ts (hunk 1 above)
Comment (dgetu) [reply packages/typespec-ts/test/util/emitUtil.ts:446]: The environment does have to be set for unit tests to consume the new code. We have expected test failures in this PR, so as tests start to pass we can look into selectively enabling it for those tests.

File: packages/typespec-ts/src/modular/serialization/serializers.ts (hunk 2 above)
Comment (dgetu) [reply packages/typespec-ts/src/modular/serialization/serializers.ts:114]: For this in particular, all of the missing functions are contained in this file at the moment. I'll open issues to track remaining work.

**Pull request merged**